tests/Unit/Test_TestCreation.cpp

```diff
@@ -0,0 +1,411 @@
+  DerivedClassWithoutMetavariables(DerivedClassWithoutMetavariables&&) =
+      default;
+  DerivedClassWithoutMetavariables& operator=(
+      DerivedClassWithoutMetavariables&&) = default;
+  ~DerivedClassWithoutMetavariables() override = default;
+
+  size_t get_value() const override { return value; }
+
+  size_t value{0};
+};
+
+struct DerivedClassWithMetavariables : BaseClass {
+  struct SizeT {
+    using type = size_t;
+    static constexpr OptionString help = {"SizeT help"};
+  };
+
+  using options = tmpl::list<SizeT>;
+  static constexpr OptionString help = {"Help"};
+
+  template <typename Metavariables>
+  // NOLINTNEXTLINE(readability-avoid-const-params-in-decls)
+  DerivedClassWithMetavariables(const size_t in_value,
+                                const OptionContext& /*context*/,
+                                Metavariables /*meta*/) {
+    value = in_value * Metavariables::value_multiplier;
+  }
+
+  DerivedClassWithMetavariables(const size_t /*in_value*/,
+                                const OptionContext& /*context*/,
+                                NoSuchType /*meta*/)
+      : value{std::numeric_limits<size_t>::max()} {}
+
+  DerivedClassWithMetavariables() = default;
+  DerivedClassWithMetavariables(const DerivedClassWithMetavariables&) = delete;
+  DerivedClassWithMetavariables& operator=(
+      const DerivedClassWithMetavariables&) = delete;
+  DerivedClassWithMetavariables(DerivedClassWithMetavariables&&) = default;
+  DerivedClassWithMetavariables& operator=(DerivedClassWithMetavariables&&) =
+      default;
+  ~DerivedClassWithMetavariables() override = default;
+
+  size_t get_value() const override { return value; }
+
+  size_t value{0};
+};
+
+void test_test_creation() {
+  // Test creation of fundamentals
+  CHECK(TestHelpers::test_creation<double>("1.846") == 1.846);
+  CHECK(
+      TestHelpers::test_creation<double, TestHelpers::TestCreationOpt<double>>(
+          "1.846") == 1.846);
+  CHECK(TestHelpers::test_creation<double, NoGroup<double>>("1.846") == 1.846);
+  CHECK(TestHelpers::test_creation<double, OneGroup<double>>("1.846") == 1.846);
+  CHECK(TestHelpers::test_creation<double, TwoGroup<double>>("1.846") == 1.846);
+
+  // Test class that doesn't need metavariables when not passing metavariables
+  CHECK(
+      TestHelpers::test_creation<ClassWithoutMetavariables,
+                                 NoGroup<ClassWithoutMetavariables>>("SizeT: 4")
+          .value == 4);
+  CHECK(TestHelpers::test_creation<ClassWithoutMetavariables,
+                                   OneGroup<ClassWithoutMetavariables>>(
+            "SizeT: 5")
+            .value == 5);
+  CHECK(TestHelpers::test_creation<ClassWithoutMetavariables,
+                                   TwoGroup<ClassWithoutMetavariables>>(
+            "SizeT: 6")
+            .value == 6);
+  CHECK(
+      TestHelpers::test_creation<ClassWithoutMetavariables>("SizeT: 7").value ==
+      7);
```

Comment:
nitpick: suggest moving this simpler case above the earlier three cases, because this could make it a bit easier to read for someone looking here to supplement the documentation.